activerecord/test/cases/adapter_test.rb

```diff
@@ -333,6 +333,118 @@ def test_select_methods_passing_a_relation
       assert_equal "special_db_type", @connection.type_to_sql(:special_db_type)
     end
 
+    def test_simple_query_with_splat
+      result = @connection.simple_query("SELECT * FROM posts WHERE id IN (:id, :id_2)", id: 1, id_2: 2)
+
+      assert result.is_a?(Array)
+
+      assert_equal 2, result.size
+
+      assert_equal(
+        {
+          "id" => 1,
+          "title" => "Welcome to the weblog",
+        },
+        result.first.slice("id", "title")
+      )
+
+      assert_equal(
+        {
+          "id" => 2,
+          "title" => "So I was thinking",
+        },
+        result.second.slice("id", "title")
+      )
+    end
+
+    def test_simple_query_with_hash_with_symbol_keys
+      result = @connection.simple_query("SELECT * FROM posts WHERE id IN (:id, :id_2)", { id: 1, id_2: 2 })
+
+      assert result.is_a?(Array)
+
+      assert_equal 2, result.size
+
+      assert_equal(
+        {
+          "id" => 1,
+          "title" => "Welcome to the weblog",
+        },
+        result.first.slice("id", "title")
+      )
+
+      assert_equal(
+        {
+          "id" => 2,
+          "title" => "So I was thinking",
+        },
+        result.second.slice("id", "title")
+      )
+    end
+
+    def test_simple_query_with_hash_with_string_keys
```

Comment:
I think there is no need to differentiate between symbol vs string keys, and testing only one case is enough.

Reply:
There does seem to be a need for this

Without symbolize_keys the following:

```ruby
@connection.simple_query("SELECT * FROM posts WHERE id IN (:id, :id_2)", { "id" => 1, "id_2" => 2 })
```

returns the following error

```
ActiveRecord::PreparedStatementInvalid: missing value for :id in SELECT id FROM posts WHERE id IN (:id, :id_2)
```

Reply:
If we feel that there is no need to handle string keys whatsoever (can be a baked in limitation) then I can remove this (and would maybe add a test case which covers the failure)

Reply:
Thinking about this further I think that not allowing string keys is probably fine. The end developer should be able to figure out this limitation fairly quickly.

I've removed the `stringify_keys` call and test for string keys